modules/public/views.py

```diff
@@ -227,38 +284,23 @@ def snap_details(snap_name):
     with appropriate sanitation.
     """
 
+    errors = []
     today = datetime.datetime.utcnow().date()
     week_ago = today - relativedelta.relativedelta(weeks=1)
 
     try:
         details = api.get_snap_details(snap_name)
-    except api.InvalidResponseContent as invalid_response_content:
-        message = str(invalid_response_content)
-        flask.abort(500, message)
-    except api.ApiErrorResponse as api_error_exception:
-        if api_error_exception.errors:
-            flask.abort(api_error_exception.status, api_error_exception.errors)
+    except api.ApiResponseError as api_error_exception:
+        if hasattr(api_error_exception, 'errors'):
+            flask.abort(
```

Comment:
So, in this case, if you get an API error, you're sending the user to an error page, rather than sending them through the main "snap details" template. This is different from what you seem to be doing in the case of most other errors - where you're using the original view template, but with some `errors` in the context.

I understand that a `404` for snap details is a special case, but here you're catching all error types.

Is this deliberate? If there's a specific reason then let's discuss it. It may be that it's actually simplest to *always* send users to an error page on any API error, and then just make sure that that page is pretty.